design/CONSUMER.md

```diff
@@ -8,4 +8,28 @@ Things to consider:
 - persisting the last seen page
 - mapping events to new data/actions
 
-This library should not try to do too much but focus on tools to make it easier to fetch events and action on them.
\ No newline at end of file
+This library should not try to do too much but focus on tools to make it easier to fetch events and action on them.
+
+## How
+
+```mermaid
+sequenceDiagram
+    participant C as Consumer
+    participant S as Consumer State
+    participant P as Producer
+    C->>+P: Fetch meta data
+    P-->>-C: Return meta data
+    
+    loop Until consumer is shut down
+        C->>+S: Fetch last stored page
```

Comment:
I noticed the usage of the word `page` to sometimes confuse me. I suggest we change it to `pageNumber` to disambiguate from 'pages of events', which we can then simply call page. I've made this change in some of the code as well, and think we should use these terms consistently.

Reply:
Heya. I was actually torn on this one and used the page on purpose. Good to discuss because I see some problems. A page could be (page number + already seen event IDs on that page). That can serve as an initial filter on actual processing. This is super important when hitting the tail to have an efficient mechanism here.